mlflow/gateway/providers/utils.py

```diff
@@ -0,0 +1,45 @@
+from typing import Dict, Any
+import aiohttp
+from fastapi import HTTPException
+
+
+async def send_request(headers: Dict[str, str], base_url: str, path: str, payload: Dict[str, Any]):
+    """
+    Send an HTTP request to a specific URL path with given headers and payload.
+
+    :param headers: The headers to include in the request.
+    :param base_url: The base URL where the request will be sent.
+    :param path: The specific path of the URL to which the request will be sent.
+    :param payload: The payload (or data) to be included in the request.
+    :return: The server's response as a JSON object.
+    :raise: HTTPException if the HTTP request fails.
+    """
+    async with aiohttp.ClientSession(headers=headers) as session:
+        url = "/".join([base_url.rstrip("/"), path.lstrip("/")])
+        async with session.post(url, json=payload) as response:
+            js = await response.json()
+            try:
+                response.raise_for_status()
+            except aiohttp.ClientResponseError as e:
+                if "error" in js:
+                    detail = js.get("error", {}).get("message", e.message)
+                else:
+                    detail = "An error occurred with your request."
+                raise HTTPException(status_code=e.status, detail=detail)
```

Comment:
Can we directly `js` to detail?

Reply:
totally. That's cleaner too. I'll do a one-liner.